pkg/network/tracer/tracer.go

```diff
@@ -343,8 +343,14 @@ func (t *Tracer) addProcessInfo(c *network.ConnectionStats) {
 	addTag("version", p.Envs["DD_VERSION"])
 	addTag("service", p.Envs["DD_SERVICE"])
 
-	if p.ContainerID != "" {
-		c.ContainerID = &p.ContainerID
+	containerID, ok := p.ContainerIndex.Get().(string)
```

Comment:
I don't think there is a need to check if the cast worked. Just `p.ContainerID.Get().(string)` works. Also I think this will allocate a new string, so we may have to make `c.ContainerID` an `intern.Value` as well.